hs_collection_resource/management/commands/fix_collection_version_missing_ispartof.py

```diff
@@ -17,19 +18,85 @@ def add_arguments(self, parser):
 
     def handle(self, *args, **options):
         dry_run = options['dryrun']
+        errors = []
+        current_site = current_site_url()
         collections = BaseResource.objects.filter(resource_type="CollectionResource").all()
+        print('-' * 100)
+        print("1. Checking Collection's metadata...")
+        col_count = collections.count()
+        print(f"{col_count} collections to check")
+        n = 0
+        fixed = 0
         for collection in collections:
-            print(f"Checking collection: {collection.short_id}")
+            n = n + 1
             for res in collection.resources.all():
                 rel_values = [rel.value for rel in res.metadata.relations.filter(type=RelationTypes.isPartOf).all()]
-                print(f"Relations: {rel_values}")
                 if collection.get_citation() not in rel_values:
-                    print("IsPartOf relation missing from resource {res}.")
+                    print(f"{n}/{col_count}:For collection:{collection}, {current_site}/resource/{collection.short_id}")
+                    print(f"IsPartOf relation missing from {res}, {current_site}/resource/{res.short_id}")
                     if dry_run:
                         print("SKIPPING creating IsPartOf relation because dry_run")
                     else:
                         print("Creating IsPartOf relation...")
                         res.metadata.create_element('relation',
                                                     type=RelationTypes.isPartOf,
                                                     value=collection.get_citation())
+                    fixed = fixed + 1
+                res_citation = res.get_citation()
+                if not collection.metadata.relations.filter(type=RelationTypes.hasPart, value=res_citation).exists():
+                    print(f"hasPart relation missing from collection: "
+                          f"{collection}, {current_site}/resource/{collection.short_id}")
+                    if dry_run:
+                        print("SKIPPING creating hasPart relation because dry_run")
+                    else:
+                        collection.metadata.create_element('relation',
+                                                           type=RelationTypes.hasPart,
+                                                           value=res_citation)
+                    fixed = fixed + 1
+        print(f"{fixed} metadata relations were missing")
+
+        print('-' * 100)
+        print("2. Remove dangling isPart metadata from resources that are not included in collections")
+        resources = BaseResource.objects.all()
+        res_count = resources.count()
+        print(f"{res_count} resources to check")
+        i = 0
+        fixed = 0
+        for res in resources:
+            i = i + 1
+            isPartOf_relations = None
+            if res.metadata:
```

Comment:
I am not sure if accessing `res.metadata` would work when `res` is BaseResource. We probably have to add this line after line 65:
`res =res.get_content_model()`

But again since Relation is part of the core metadata, it probably works without the above line. I was referring to existing commands, and I find whenever we get a res as BaseResource we then use the above line so that the right metadata object can be accessed. It's a mezzanine thing probably. But if you have found the command to work fine, then ignore this comment.

Reply:
It seems like it works okay...